front-end/src/app/layout/sidebar/menus/abstract-menu.component.ts

```diff
@@ -1,41 +1,50 @@
 import { Component, OnInit } from '@angular/core';
 import { DestroyerComponent } from '../../../shared/components/app-destroyer.component';
 import { selectActiveReport } from '../../../store/active-report.selectors';
-import { combineLatest, Observable, of, switchMap, takeUntil } from 'rxjs';
-import { selectSidebarState } from '../../../store/sidebar-state.selectors';
+import { filter, map, Observable, of, takeUntil } from 'rxjs';
 import { ReportSidebarSection, SidebarState } from '../sidebar.component';
 import { Report } from '../../../shared/models/report.model';
 import { MenuItem } from 'primeng/api';
 import { Store } from '@ngrx/store';
 import { ReportService } from '../../../shared/services/report.service';
+import { collectRouteData } from 'app/shared/utils/route.utils';
+import { NavigationEnd, Router } from '@angular/router';
 
 @Component({
   template: '',
 })
 export abstract class AbstractMenuComponent extends DestroyerComponent implements OnInit {
   activeReport$?: Observable<Report | undefined>;
+  activeReport: Report | undefined;
   items$: Observable<MenuItem[]> = of([]);
   reportString?: string;
+  sidebarState: SidebarState | undefined;
 
   protected constructor(
     private store: Store,
     private reportService: ReportService,
+    private router: Router,
   ) {
     super();
   }
 
   ngOnInit(): void {
     this.activeReport$ = this.store.select(selectActiveReport);
+    this.sidebarState = new SidebarState(collectRouteData(this.router)['sidebarSection']);
 
-    this.items$ = combineLatest([this.store.select(selectSidebarState), this.activeReport$]).pipe(
+    this.items$ = this.activeReport$.pipe(
       takeUntil(this.destroy$),
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      switchMap(([sidebarState, activeReport]: [SidebarState, Report | undefined]) => {
-        const isEditable = this.reportService.isEditable(activeReport);
-
-        return of(this.getMenuItems(sidebarState, activeReport, isEditable));
+      map((activeReport: Report | undefined) => {
+        this.activeReport = activeReport;
+        return this.updateMenuItems(activeReport);
       }),
     );
+
+    this.router.events.pipe(filter((event) => event instanceof NavigationEnd)).subscribe(() => {
```

Comment:
you're going to want to combineLatest the router even and the active report observables.  when we take data out of an observable and put it into a synchronous variable like `this.activeReport` or `this.sidebarState` we forfeit control of flow.  so later in `this.getMenuItems(this.sidebarState as SidebarState, activeReport, isEditable);` we don't know whether sidebarState is ready or not.

I'd recommend using a combineLatest like this:
```
const routerEvents$ = this.router.events.pipe(filter((event) => event instaceof NavigationEnd));
this.items$ = combineLatest([this.activeReport$, routerEvents$]).pipe.map(([activeReport, sidebarState]) => {
    const isEditable = this.reportService.isEditable(activeReport);
    return this.getMenuItems(this.sidebarState as SidebarState, activeReport, isEditable);
});
```
then `activeReport` and `sidebarState` are no longer needed and don't pose any risk of being used when their data is not up to date